File: Gherkin.Specs/TestTokenFormatter.cs
using System;
using System.Linq;

namespace Gherkin.Specs
{
	public class TestTokenFormatter
	{
		public string FormatToken(Token token)
		{
			if (token.IsEOF)
				return "EOF";

			return string.Format("({0}:{1}){2}:{3}/{4}/{5}", 
				token.Line.LineNumber,
				token.MatchedIndent + 1,
				token.MatchedType,
				token.MatchedKeyword,
				token.MatchedText,
				token.MathcedItems == null ? "" : string.Join(",", token.MathcedItems.Select(i => i.Column + ":" + i.Text))
				);
		}
	}
}

File: java/src/test/java/gherkin/TestTokenFormatter.java
package gherkin;

public class TestTokenFormatter {
    private static final StringUtils.ToString<GherkinLineSpan> SPAN_TO_STRING = new StringUtils.ToString<GherkinLineSpan>() {
        @Override
        public String toString(GherkinLineSpan o) {
            return o.Column + ":" + o.Text;
        }
    };

    public String FormatToken(Token token) {
        if (token.IsEOF())
            return "EOF";

        return String.format("(%s:%s)%s:%s/%s/%s",
                token.Location.Line,
                token.Location.Column,
                token.MatchedType,
                token.MatchedKeyword,
                token.MatchedText,
                token.MathcedItems == null ? "" : StringUtils.join(SPAN_TO_STRING, ",", token.MathcedItems)
        );
    }
}
